src/components/DatePicker.js

```diff
@@ -0,0 +1,454 @@
+    super(props);
+    this.state = {
+      currentDate: null,
+      showCalendar: false
+    };
+  }
+
+  _shouldBorderCell (cellCount) {
+    return cellCount % this.props.calendarColumns !== 0;
+  }
+
+  _shouldBorderRow (currentCell, cellCount) {
+    const rows = (cellCount / this.props.calendarColumns) - 1;
+
+    return (currentCell / this.props.calendarColumns) < rows;
+  }
+
+  _handleDateSelect (date) {
+    if (this.props.closeOnDateSelect) {
+      this._handleBlur();
+    }
+
+    this.props.onDateSelect(date);
+  }
+
+  _handlePreviousClick () {
+    const selectedDate = moment.unix(this.props.selectedDate).locale(this.props.locale);
+    let currentDate = this.state.currentDate ? this.state.currentDate.locale(this.props.locale) : selectedDate;
+
+    currentDate = moment(currentDate.startOf('month').subtract(1, 'm'), this.props.format);
+
+    this.setState({
+      currentDate
+    });
+  }
+
+  _handleNextClick () {
+    const selectedDate = moment.unix(this.props.selectedDate).locale(this.props.locale);
+    let currentDate = this.state.currentDate ? this.state.currentDate.locale(this.props.locale) : selectedDate;
+
+    currentDate = moment(currentDate.endOf('month').add(1, 'd'), this.props.format);
+
+    this.setState({
+      currentDate
+    });
+  }
+
+  _renderMonthTable (styles, currentDate, selectedDate) {
+    const days = [];
+    const startDate = moment(currentDate, this.props.format).startOf('month').startOf('week');
+    const endDate = moment(currentDate, this.props.format).endOf('month').endOf('week');
+    const minimumDate = this.props.minimumDate ? moment(this.props.minimumDate, this.props.format) : null;
+
+    while (startDate.isBefore(endDate)) {
+      const isCurrentMonth = startDate.month() === currentDate.month();
+      const isCurrentDay = startDate.format(this.props.format) === selectedDate.format(this.props.format);
+      const day = startDate.isBefore(minimumDate) ? (
+        <div
+          key={startDate.month() + '-' + startDate.date()}
+          style={styles.calendarDay}
+        >
+          <div
+            key={startDate.format('DDDD')}
+            style={[styles.calendarDayContent, styles.calendarDayDisabled]}
+          >
+            {startDate.date()}
+          </div>
+        </div>
+      ) : (
+        <div
+          key={startDate.month() + '-' + startDate.date()}
+          onClick={this._handleDateSelect.bind(this, startDate.unix())}
+          style={[
+            styles.calendarDay,
+            isCurrentMonth && styles.currentMonth
+          ]}
+        >
+          <div
+            key={startDate.format('DDDD')}
+            style={[styles.calendarDayContent, isCurrentDay && styles.currentDay]}
+          >
+            <div style={styles.calendarDayText}>{startDate.date()}</div>
+          </div>
+        </div>
+      );
+
+      days.push(day);
+      startDate.add(1, 'd');
+    }
+
+    //Divide calendar columns/Rows
+    if (this.props.divideCells) {
```

Comment:
We are already looping through the days above in the `while` statement. Is there a way we can move this logic into that statement? At first glance it looks like we could use Radium to apply the appropriate styles.